spring-integration-jdbc/src/test/java/org/springframework/integration/jdbc/StoredProcJavaConfigTests.java

```diff
@@ -0,0 +1,122 @@
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThat;
+
+import java.sql.Types;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import javax.sql.DataSource;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.integration.annotation.InboundChannelAdapter;
+import org.springframework.integration.annotation.Poller;
+import org.springframework.integration.channel.QueueChannel;
+import org.springframework.integration.config.EnableIntegration;
+import org.springframework.integration.core.MessageSource;
+import org.springframework.integration.jdbc.storedproc.PrimeMapper;
+import org.springframework.integration.jdbc.storedproc.ProcedureParameter;
+import org.springframework.jdbc.core.SqlParameter;
+import org.springframework.jdbc.datasource.embedded.EmbeddedDatabaseBuilder;
+import org.springframework.jdbc.datasource.embedded.EmbeddedDatabaseType;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.PollableChannel;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+/**
+ * Equivalent to {@link StoredProcPollingChannelAdapterWithNamespaceIntegrationTests}.
+ *
+ * @author Gary Russell
+ * @since 4.2
+ *
+ */
+@ContextConfiguration(classes = StoredProcJavaConfigTests.Config.class)
+@RunWith(SpringJUnit4ClassRunner.class)
+@DirtiesContext
+public class StoredProcJavaConfigTests {
+
+	@Autowired
+	private PollableChannel fooChannel;
+
+	@Test
+	public void test() {
+		Message<?> received = fooChannel.receive(10000);
+		assertNotNull(received);
+		Collection<?> primes = (Collection<?>) received.getPayload();
+		assertThat(primes, contains(2, 3, 5, 7));
+		received = fooChannel.receive(100);
+		assertNull(received);
+	}
+
+	@Configuration
+	@EnableIntegration
+	public static class Config {
+
+		@Bean
+		public PollableChannel fooChannel() {
+			return new QueueChannel();
+		}
+
+		@Bean
+		@InboundChannelAdapter(value = "fooChannel", poller = @Poller(fixedDelay="5000", maxMessagesPerPoll = "1"))
+		public MessageSource<?> storedProc() {
+			StoredProcPollingChannelAdapter source = new StoredProcPollingChannelAdapter(storedProcExecutor());
+			source.setExpectSingleResult(true);
+			return source;
+		}
+
+		@Bean
+		public StoredProcExecutor storedProcExecutor() {
+			StoredProcExecutor executor = new StoredProcExecutor(dataSource());
+			executor.setIgnoreColumnMetaData(true);
+			executor.setStoredProcedureName("GET_PRIME_NUMBERS");
+			List<ProcedureParameter> procedureParameters = new ArrayList<>();
```

Comment:
Java 7 (and above) is allowed only in the `core` module. 
Will be fixed during merge